dotCMS/src/main/java/com/dotmarketing/portlets/workflows/ajax/WfStepAjax.java

```diff
@@ -81,23 +82,56 @@ public void reorder(HttpServletRequest request, HttpServletResponse response) th
 		}
 			
 	}
-	
-	public void delete(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-		String stepId = request.getParameter("stepId");
 
-		WorkflowAPI wapi = APILocator.getWorkflowAPI();
-		
+	public void delete(final HttpServletRequest request,
+					   final HttpServletResponse response) throws ServletException, IOException {
+
+		final String stepId = request.getParameter("stepId");
+
 		try {
 
-			WorkflowStep step = wapi.findStep(stepId);
-			wapi.deleteStep(step);
-		} catch (DotDataException e) {
+			this.workflowHelper.deleteStep (stepId);
+		} catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)